docs/user_guide/storing/docindex.md

```diff
@@ -37,716 +37,75 @@ Currently, DocArray supports the following vector databases:
-
-If a Document contains a DocList, it can still be stored in a Document Index.
-In this case, the DocList will be represented as a new index (or table, collection, etc., depending on the database backend), that is linked with the parent index (table, collection, ...).
-
-This still lets index and search through all of your data, but if you want to avoid the creation of additional indexes you could try to refactor your document schemas without the use of DocList.
-
-
-**Index**
-
-In the following example you can see a complex schema that contains nested Documents with subindex.
-The `MyDoc` contains a `DocList` of `VideoDoc`, which contains a `DocList` of `ImageDoc`, alongside some "basic" fields:
-
-```python
-class ImageDoc(BaseDoc):
-    url: ImageUrl
-    tensor_image: AnyTensor = Field(space='cosine', dim=64)
-
-
-class VideoDoc(BaseDoc):
-    url: VideoUrl
-    images: DocList[ImageDoc]
-    tensor_video: AnyTensor = Field(space='cosine', dim=128)
-
-
-class MyDoc(BaseDoc):
-    docs: DocList[VideoDoc]
-    tensor: AnyTensor = Field(space='cosine', dim=256)
-
-
-# create a Document Index
-doc_index = HnswDocumentIndex[MyDoc](work_dir='/tmp3')
-
-# create some data
-index_docs = [
-    MyDoc(
-        docs=DocList[VideoDoc](
-            [
-                VideoDoc(
-                    url=f'http://example.ai/videos/{i}-{j}',
-                    images=DocList[ImageDoc](
-                        [
-                            ImageDoc(
-                                url=f'http://example.ai/images/{i}-{j}-{k}',
-                                tensor_image=np.ones(64),
-                            )
-                            for k in range(10)
-                        ]
-                    ),
-                    tensor_video=np.ones(128),
-                )
-                for j in range(10)
-            ]
-        ),
-        tensor=np.ones(256),
-    )
-    for i in range(10)
-]
-
-# index the Documents
-doc_index.index(index_docs)
-```
-
-**Search**
-
-You can perform search on any subindex level by using `find_subindex()` method and the dunder operator `'root__subindex'` to specify the index to search on.
-
-```python
-# find by the `VideoDoc` tensor
-root_docs, sub_docs, scores = doc_index.find_subindex(
-    np.ones(128), subindex='docs', search_field='tensor_video', limit=3
-)
-
-# find by the `ImageDoc` tensor
-root_docs, sub_docs, scores = doc_index.find_subindex(
-    np.ones(64), subindex='docs__images', search_field='tensor_image', limit=3
+# Perform a hybrid search - combining vector search with filtering
+query = (
+    doc_index.build_query()  # get empty query object
+    .find(query=np.ones(128), search_field='embedding')  # add vector similarity search
+    .filter(filter_query={'price': {'$gte': 2}})  # add filter search
+    .build()  # build the query
 )
+retrieved_docs, scores = doc_index.execute_query(query)
 ```
```

Comment:
I think we should here again add a big fat link to all the backend documentation pages and tell people that they can get more detailed information there